src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -0,0 +1,123 @@
+
+        val violations = seq.map(item => (item, validator(item))).zipWithIndex.collect {
+          case ((item, f: Failure), pos: Int) => GroupViolation(item, "not valid", Some(s"[$pos]"), f.violations)
+        }
+
+        if (violations.isEmpty) Success
+        else Failure(Set(GroupViolation(seq, "seq contains elements, which are not valid", None, violations.toSet)))
+      }
+    }
+  }
+
+  implicit lazy val failureWrites: Writes[Failure] = Writes { f =>
+    Json.obj("errors" -> f.violations.flatMap(allRuleViolationsWithFullDescription(_)))
+  }
+
+  implicit lazy val ruleViolationWrites: Writes[RuleViolation] = Writes { v =>
+    Json.obj(
+      "attribute" -> v.description,
+      "error" -> v.constraint
+    )
+  }
+
+  private def allRuleViolationsWithFullDescription(violation: Violation,
+                                 parentDesc: Option[String] = None,
+                                 attachDot: Boolean = true): Set[RuleViolation] = {
+    def concatPath(parent: String, child: Option[String], attachDot: Boolean): String = {
+      child.map(c => parent + { if(attachDot) "." else "" } + c).getOrElse(parent)
+    }
+
+    violation match {
+      case r: RuleViolation => Set(parentDesc.map(p =>
+        r.withDescription(concatPath(p, r.description, attachDot)))
+        .getOrElse(r))
+      case g: GroupViolation => g.children.flatMap { c =>
+        parentDesc.map { p =>
+          val desc = concatPath(p, g.description, attachDot)
+          allRuleViolationsWithFullDescription(c, Some(desc), g.description.isDefined)
+        } getOrElse {
+          allRuleViolationsWithFullDescription(c, g.description, g.description.isDefined)
+        }
+      }
+    }
+  }
+
+  def urlsCanBeResolvedValidator: Validator[String] = {
+    new Validator[String] {
+      def apply(url: String) = {
+        Try {
+          new URL(url).openConnection() match {
+            case http: HttpURLConnection =>
+              http.setRequestMethod("HEAD")
+              if (http.getResponseCode == HttpURLConnection.HTTP_OK) Success
+              else Failure(Set(RuleViolation(url, "url could not be resolved", None)))
+            case other: URLConnection =>
+              other.getInputStream
+              Success //if we come here, we could read the stream
+          }
+        }.getOrElse(
+          Failure(Set(RuleViolation(url, "url could not be resolved", None)))
+        )
+      }
+    }
+  }
+
+  def elementsAreUnique[A]: Validator[Seq[A]] = {
+    new Validator[Seq[A]] {
+      def apply(seq: Seq[A]) = {
+        if (seq.size == seq.distinct.size) Success
+        else Failure(Set(RuleViolation(seq, "Elements must be unique", None)))
+      }
+    }
+  }
+
+  case class ViolationMessageAndProperty(message: String, property: Option[String]) {
+    override def toString: String = s"Property: $property Message: $message"
+  }
+
+  def getAllRuleConstrains(r: Result): Seq[ViolationMessageAndProperty] = {
```

Comment:
Do you mean `elementsAreUnique` method? For the time being it is only used once in `V2AppDefinition`. I am moving it there in the following commit. Just for future reference - should I avoid defining general Validators?